docs: GitHub actions (#3237)
lightdash/lightdash: docs/docs/guides/cli/how-to-automatically-deploy.mdx

> # How to automatically deploy 

Reviewer: ```suggestion
# How to automatically sync your change in Lightdash
```

I'm not sure if we should also add `automatically deploy` somewhere in the title too so it's more searchable? 

I just think as a data person `automatically deploy` isn't a word I'd be looking for (it's more "syncing" that I'd be trying to find)

> # How to automatically sync your change in Lightdash

Reviewer: ```suggestion
# How to automatically sync your changes in Lightdash
```

> If you use github to store your DBT config, now you can setup a `github action` to make it deploy automatically 

Reviewer: ```suggestion
If you've connected Lightdash to GitHub, you can setup a `github action` to make Lightdash deploy your project automatically.
```

the new config into Lightdash.

## First add the credentials to Github secrets

> We are going to add some secrets and config to github actions, 

Reviewer: ```suggestion
We are going to add some secrets and config to GitHub actions, 
```

> but you don't want those to be public , so the best way is to add it as secrets on Github

Reviewer: ```suggestion
but you don't want those to be public , so the best way to do this is to add them as secrets on Github
```

Go to your repo, click on `Settings` , on the left sidebar, click on `Secrets` under `Security`. 
Now click on the `New repository secret`

![Github secrets page](./assets/github-secrets.png)

We need the following credentials, you can find them on the app, 
or in your `~/.config/lightdash/config.yaml` file on your machine if you have used the Lightdash CLI already.

- `LIGHTDASH_API_KEY`: Your personal access token on Lightdash. [See these instructions](https://docs.lightdash.com/guides/cli/cli-authentication#:~:text=First%2C%20you%27ll%20need%20to%20create%20a%20new%20personal%20access%20token%20in%20the%20UI%20by%20going%20to%20Settings%20%3E%20Personal%20Access%20Tokens.%20You%20can%27t%20use%20an%20existing%20personal%20access%20token!%20You%20have%20to%20create%20a%20new%20one%20just%20for%20yourself.) for creating a personal access token in the app.
- `LIGHTDASH_PROJECT`: The UUID for your project. For example, if your URL looks like `https://eu1.lightdash.cloud/projects/3538ab33-dc90-aabb-bc00-e50bba3a5f69/tables`, then `3538ab33-dc90-45f0-aabb-e50bba3a5f69` is your `LIGHTDASH_PROJECT`  
- `LIGHTDASH_URL`: (like https://eu1.lightdash.cloud or https://app.lightdash.cloud)

> - `DBT_PROFILES`: Copy your full profiles.yml file into this secret, like:

Reviewer: What happens if I have a dev and a prod target here? (but dev is default?)

Do I need to adjust my `profiles.yml` to only have my prod target? So that the GitHub action is deploying the production version of my dbt project?

Author: you could copy a modified `profiles` file to just release the one you want, or you could add a `--profile <name>` property to the CLI command on github actions.

Reviewer: nice!

Where would you add the `--profile <name>` in the GitHub actions bit? 

I think it would be great to add this to the instructions :) But happy to do it after we've merged this!
